lib/cli.rb

```diff
@@ -529,49 +518,71 @@ def self.check_port_validity(port)
+    name = options[:name] || host
 
 
-      if !scope_list.empty?
-        inspected_scopes = " for #{Machinery::Ui.internal_scope_list_to_string(scope_list)}"
-      end
-      Machinery::Ui.puts "Inspecting #{host}#{inspected_scopes}..."
+    if !scope_list.empty?
+      inspected_scopes = " for #{Machinery::Ui.internal_scope_list_to_string(scope_list)}"
+    end
+    Machinery::Ui.puts "Inspecting #{host}#{inspected_scopes}..."
 
-      inspect_options = {}
-      if options["show"]
-        inspect_options[:show] = true
-      end
-      if options["verbose"]
-        inspect_options[:verbose] = true
-      end
-      if options["extract-files"] || options["extract-changed-config-files"]
-        inspect_options[:extract_changed_config_files] = true
-      end
-      if options["extract-files"] || options["extract-changed-managed-files"]
-        inspect_options[:extract_changed_managed_files] = true
-      end
-      if options["extract-files"] || options["extract-unmanaged-files"]
-        inspect_options[:extract_unmanaged_files] = true
-      end
-      inspect_options[:remote_user] = options["remote-user"]
+    inspect_options = {}
+    if options["show"]
+      inspect_options[:show] = true
+    end
+    if options["verbose"]
+      inspect_options[:verbose] = true
+    end
+    if options["extract-files"] || options["extract-changed-config-files"]
+      inspect_options[:extract_changed_config_files] = true
+    end
+    if options["extract-files"] || options["extract-changed-managed-files"]
+      inspect_options[:extract_changed_managed_files] = true
+    end
+    if options["extract-files"] || options["extract-unmanaged-files"]
+      inspect_options[:extract_unmanaged_files] = true
+    end
 
-      filter = FilterOptionParser.parse("inspect", options)
+    filter = FilterOptionParser.parse("inspect", options)
 
-      if options["verbose"] && !filter.empty?
-        Machinery::Ui.puts "\nThe following filters are applied during inspection:"
-        Machinery::Ui.puts filter.to_array.join("\n") + "\n\n"
-      else
-        show_filter_note(scope_list, filter)
-      end
+    if options["verbose"] && !filter.empty?
+      Machinery::Ui.puts "\nThe following filters are applied during inspection:"
+      Machinery::Ui.puts filter.to_array.join("\n") + "\n\n"
+    else
+      show_filter_note(scope_list, filter)
+    end
+
+    [name, scope_list, inspect_options, filter]
+  end
+
+  desc "Inspect running system"
+  long_desc <<-LONGDESC
+    Inspect running system and generate system descripton from inspected data.
+
+    Multiple scopes can be passed as comma-separated list. If no specific scopes
+    are given, all scopes are inspected.
+
+    Available scopes: #{AVAILABLE_SCOPE_LIST}
+  LONGDESC
+  arg "HOSTNAME"
+  command "inspect" do |c|
+    supports_filtering(c)
+    define_inspect_command_options(c)
+
+    c.action do |global_options,options,args|
```

Comment:
Unused block argument - `global_options`. If it's necessary, use `_` or `_global_options` as an argument name to indicate that it won't be used.<br>Space missing after comma.